src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,171 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    public static final String TIME_ZONE = "+02:00";
+    public static final String TIME_FORMATTER_FORMAT_DATE = "dd MMMM yyyy HH:mm";
+    public static final String TIME_FORMATTER_CUSTOM_PARSE_DATE = "d MMM yyyy";
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate date = LocalDate.now();
+        switch (datePart) {
+            case FULL:
+                return String.valueOf(date);
+            case YEAR:
+                return String.valueOf(date.getYear());
+            case MONTH:
+                return String.valueOf(date.getMonth());
+            case DAY:
+                return String.valueOf(date.getDayOfMonth());
+            default:
+                throw new DateTimeException("Can't get date!");
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.ofNullable(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
```

Comment:
```suggestion
            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
```
we are sure that we put to this Optional some value

Reply:
done